src/Extension/Publishable/PublishableSiteTree.php

```diff
@@ -90,6 +90,23 @@ public function objectsToDelete($context)
      */
     public function urlsToCache()
     {
-        return [Director::absoluteURL($this->getOwner()->Link()) => 0];
+        $link = $this->getOwner()->Link();
+        if (strpos($link, '?') === false) {
```

Comment:
Use `parse_url($url, PHP_URL_QUERY)` to extract the query string rather than doing it yourself - https://www.php.net/manual/en/function.parse-url.php

Reply:
Thanks for the tip, used a different approach.